code/datums/singularities.dm

```diff
@@ -0,0 +1,310 @@
+/*
+Singularity behaviors, for all your singularity needs.
+
+This file contains the base type and:
+- Normal Singularity
++ Katamari (normal singulo sub-type)
+*/
+
+///
+
+#define EVENT_GROWTH 3//the rate at which the event proc radius is scaled relative to the radius of the singularity
+#define EVENT_MINIMUM 5//the base value added to the event proc radius, serves as the radius of a 1x1
+
+ABSTRACT_TYPE(/datum/singularity_behavior/)
+/datum/singularity_behavior/
+	/// What singularity object is using me?
+	var/obj/machinery/the_singularity/singularity
+	/// Use the warpy effect from modern singularity. Props to amylizzle!
+	var/use_amylizzle_animation = TRUE
+	/// Set this to a valid icon file to override the singularity icon.
+	var/special_icon
+	/// This will override the singularity icon_state.
+	var/special_icon_state
+
+/datum/singularity_behavior/New(var/obj/machinery/the_singularity/owner, var/datum/singularity_behavior/old_type)
+	. = ..()
+	src.singularity = owner
+	if (use_amylizzle_animation)
+		//get all bendy
+
+		var/image/lense = image(icon='icons/effects/overlays/lensing.dmi', icon_state="lensing_med_hole", pixel_x = -208, pixel_y = -208)
+		lense.plane = PLANE_DISTORTION
+		lense.blend_mode = BLEND_OVERLAY
+		lense.appearance_flags = RESET_ALPHA | RESET_COLOR
+		src.singularity.UpdateOverlays(lense, "grav_lensing")
+
+
+/// Called when an explosive is powerful enough to kill the normal singularity
+/datum/singularity_behavior/proc/explosive_kill(var/severity, var/last_touched, var/power)
+	return
+
+/// Called when the singularity attempts to consume an atom
+/datum/singularity_behavior/proc/consume_atom(var/atom/A, no_visuals = FALSE)
+	return
+
+/// Called when the singularity is supposed to animate the atom being consumed
+/datum/singularity_behavior/proc/animate_consume_atom(var/atom/A)
+	return
+
+/// Called when the singularity wants to move it move it
+/datum/singularity_behavior/proc/move()
+	return
+
+/// Called when the singularity wants to grow
+/datum/singularity_behavior/proc/grow(var/severity)
+	return
+
+/// Random events the singularity does occasionaly
+/datum/singularity_behavior/proc/event()
+	return
+
+// qdeling the behavior itself makes it not gc, and to avoid race conditions,
+// i'm not nullifying the behavior on the singularity.
+
+/datum/singularity_behavior/disposing()
+	src.singularity = null
```

Comment:
I feel like you should remove the distortion thing here if you added it in New.

Reply:
oh, right, this reminds me to think up of a clean way to GC the behavior without deleting the singularity, which is one of the main points behind the design of the datum behavior.